### PR TITLE
Add support for `inspect`

### PR DESCRIPTION
**Pull request by victorb**
Returned output is wildly incomplete right now

License: MIT
Signed-off-by: Victor Bjelkholm <git@victor.earth>

### CODE REVIEW

File: test/clj_docker_client/core_test.clj [hunk 1]
```diff
@@ -120,6 +126,17 @@
...
+(deftest test-inspect
+  (with-open [conn ^DockerClient (connect)]
+    (pull conn img)
+    (testing "Inspecting a container"
+      (let [container-id (create conn img "echo ok" {} {})]
+        (start conn container-id)
+        (let [inspect-res (inspect conn container-id)
+              ip (-> inspect-res :network-settings :ip-address)]
+          (is (not (nil? ip)))
+          (is (correct-ip? ip)))))))
```
Comment (lispyclouds) [inline test/clj_docker_client/core_test.clj]: Remove the container and the pulled image at the end of the test. See https://github.com/lispyclouds/clj-docker-client/blob/master/test/clj_docker_client/core_test.clj#L120-L121 Alternatively, can do this test in the `test-containers` test.

File: src/clj_docker_client/formatters.clj [hunk 2]
```diff
@@ -212,3 +214,13 @@
...
+
+(defn format-inspect-network [^NetworkSettings network-settings]
+  {:ip-address (.ipAddress network-settings)
+   :ip-prefix-len (.ipPrefixLen network-settings)})
+
+;; TODO missing bunch of interesting fields here
+;; See https://github.com/spotify/docker-client/blob/6580ae844a3be96421dfce78707163526c8ba82c/src/main/java/com/spotify/docker/client/messages/ContainerInfo.java#L147
+(defn format-inspect
+  [^ContainerInfo container-info]
+  {:network-settings (format-inspect-network (.networkSettings container-info))})
```
Comment (lispyclouds) [inline src/clj_docker_client/formatters.clj]: Try adding the other attributes. Things like formatting the container state is already available in https://github.com/lispyclouds/clj-docker-client/blob/master/src/clj_docker_client/formatters.clj#L65

File: test/clj_docker_client/core_test.clj (hunk 1 above)
Comment (victorb) [reply test/clj_docker_client/core_test.clj]: Thanks, fixed!

File: src/clj_docker_client/formatters.clj (hunk 2 above)
Comment (victorb) [reply src/clj_docker_client/formatters.clj]: I started solving this but quickly end up tired of just manually rewriting fields from `.hostConfig` to `:host-config` and so on. Has to be a better way out there than manually converting the fields and locking ourselves down to the output.

What about a function that can format the entire output by going through all the public java fields (or getters I guess), making the kebab-case into dash-separated keywords?

Comment (lispyclouds) [reply src/clj_docker_client/formatters.clj]: Thats actually a good idea. I would look into it.

File: test/clj_docker_client/core_test.clj [hunk 3]
```diff
@@ -15,6 +15,7 @@
 
 (ns clj-docker-client.core-test
   (:require [clojure.test :refer :all]
+            [clojure.pprint :refer [pprint]]
```
Comment (lispyclouds) [inline test/clj_docker_client/core_test.clj]: this is unused, but I will remove it myself later.

**Pull request merged**